react/src/components/VFolderNodeDescription.tsx

```diff
@@ -0,0 +1,345 @@
+import Flex from './Flex';
+import { statusTagColor } from './VFolderNodes';
+import { VFolderNodeDescriptionFragment$key } from './__generated__/VFolderNodeDescriptionFragment.graphql';
+import { VFolderNodeDescriptionPermissionRefreshQuery } from './__generated__/VFolderNodeDescriptionPermissionRefreshQuery.graphql';
+import { CheckCircleOutlined, UserOutlined } from '@ant-design/icons';
+import {
+  App,
+  Descriptions,
+  theme,
+  Tooltip,
+  Typography,
+  type DescriptionsProps,
+} from 'antd';
+import { createStyles } from 'antd-style';
+import graphql from 'babel-plugin-relay/macro';
+import dayjs from 'dayjs';
+import _ from 'lodash';
+import { useTranslation } from 'react-i18next';
+import { fetchQuery, useFragment, useRelayEnvironment } from 'react-relay';
+
+const useStyle = createStyles(({ css }) => {
+  return {
+    path: css`
+      width: fit-content;
+      max-width: none;
+      @media (max-width: 280px) {
+        max-width: 150px;
+      }
+    `,
+  };
+});
+
+interface VFolderNodeDescriptionProps extends DescriptionsProps {
+  vfolderNodeFrgmt?: VFolderNodeDescriptionFragment$key | null;
+  onRequestRefresh?: () => void;
+}
+
+const VFolderNodeDescription: React.FC<VFolderNodeDescriptionProps> = ({
+  vfolderNodeFrgmt,
+  onRequestRefresh,
+  ...props
+}) => {
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { styles } = useStyle();
+  const { message } = App.useApp();
+
+  const relayEnv = useRelayEnvironment();
+  const currentProject = useCurrentProjectValue();
+  const userRole = useCurrentUserRole();
+  const painKiller = usePainKiller();
+  const baiClient = useSuspendedBackendaiClient();
+  const [currentUser] = useCurrentUserInfo();
+
+  const updateMutation = useTanMutation({
+    mutationFn: ({ permission, id }: { permission: string; id: string }) => {
+      return baiClient.vfolder.update_folder({ permission }, id);
+    },
+  });
+
+  const vfolderNode = useFragment(
+    graphql`
+      fragment VFolderNodeDescriptionFragment on VirtualFolderNode {
+        id
+        host
+        quota_scope_id
+        user
+        user_email
+        group
+        group_name
+        creator
+        usage_mode
+        permission
+        ownership_type
+        max_files
+        max_size
+        created_at
+        last_used
+        num_files
+        cur_size
+        cloneable
+        status
+        permissions @since(version: "24.09.0")
+        ...VFolderPermissionCellFragment
+      }
+    `,
+    vfolderNodeFrgmt,
+  );
+
+  if (!vfolderNode) {
+    return null;
+  }
+
+  const quotaScopeId = _.split(vfolderNode?.quota_scope_id, ':')?.[1];
+  const vfolderId = toLocalId(vfolderNode?.id);
+  const [vfolderIdPrefix1, vfolderIdPrefix2, ...vfolderIdRest] = [
+    vfolderId.slice(0, 2),
+    vfolderId.slice(2, 4),
+    vfolderId.slice(4),
+  ];
+
+  const items: DescriptionsProps['items'] = filterEmptyItem([
+    {
+      key: 'path',
+      label: (
+        <Tooltip
```

Comment:
Compositions of components in the props make me feel verbose and less readability. Can we make it to separated components? When it comes to code reading, the readabilty is not good for me. Because I need to read all of code details in the props to understand. but not a big deal. eg) <PathLable>